src/Adapter/LegacyContext.php

```diff
@@ -282,7 +282,7 @@ public function getEmployeeLanguageIso()
     /**
      * Returns Currency set for the current employee.
      *
-     * @return Currency
+     * @return string
```

Comment:
Ok there is something clearly wrong here!!
First Currency::sign is deprecated so it shouldn't be used, symbol is the right field

Second `getEmployeeCurrency` is only used in `LayoutExtension` and it initializes default twig variables, but the fallback when this method returns nothing is `$this->currencyDataProvider->getDefaultCurrency()` which returns a currency object

So either:
- both should return a currency and the LayoutExtension gets the symbol field to insert into twig
- or this one is correct with string output but then the fallback on default currency should use the symbol field not the Currency itself

Reply:
By the way I'm the one who messed it up 😅

Reply:
I checked the PR that modified this https://github.com/PrestaShop/PrestaShop/pull/12046
So the string return is expected, so you should fix the LayoutExtension using `$this->currencyDataProvider->getDefaultCurrency()->getSymbol()` as the fallback

Reply:
And `getSymbol` should also be used in `getEmployeeCurrency` instead of `->sign`

Reply:
Ok. This method `getEmployeeCurrency` returns now the Currency object. And in `LayoutExtension`, I get the `symbol' ;)

Reply:
Ping @PierreRambaud